playground/backend/internal/fs_tool/go_fs_test.go

```diff
@@ -0,0 +1,81 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fs_tool
+
+import (
+	"fmt"
+	"github.com/google/uuid"
+	"reflect"
+	"testing"
+)
+
+func Test_newGoLifeCycle(t *testing.T) {
+	pipelineId := uuid.New()
+	workingDir := "workingDir"
+	baseFileFolder := fmt.Sprintf("%s/%s/%s", workingDir, baseFileFolder, pipelineId)
+	srcFileFolder := baseFileFolder + "/src"
+	binFileFolder := baseFileFolder + "/bin"
+
+	type args struct {
+		pipelineId uuid.UUID
+		workingDir string
+	}
+	tests := []struct {
+		name string
+		args args
+		want *LifeCycle
+	}{
+		{
+			// Test case with calling newGoLifeCycle method with correct pipelineId and workingDir.
+			// As a result, want to receive an expected go life cycle.
+			name: "newGoLifeCycle",
```

Comment:
Do we have a negative test case here?

Reply:
This is a constructor and I think we cannot write a negative test case here